interfaces/apparmor/spec.go

```diff
@@ -160,6 +160,31 @@ func (spec *Specification) AddSnapLayout(si *snap.Info) {
 	}
 }
 
+// AddParallelInstanceMapping adds AppArmor snippets allowing remapping of snap
+// directories for parallel installed snaps
+//
+// Specifically snap-update-ns will apply the following bind mounts
+// - /snap/foo_bar -> /snap/foo
+// - /var/snap/foo_bar -> /var/snap/foo
+// - /home/joe/snap/foo_bar -> /home/joe/snap/foo
+func (spec *Specification) AddParallelInstanceMapping(si *snap.Info) {
+	if si.InstanceKey == "" {
+		return
+	}
+	var buf bytes.Buffer
+
+	// /snap/foo_bar -> /snap/foo
+	fmt.Fprintf(&buf, "  # Allow parallel instance snap mount namespace adjustments\n")
+	fmt.Fprintf(&buf, "  mount options=(rw rbind) /snap/%s/ -> /snap/%s/,\n", si.InstanceName(), si.SnapName())
+	// /var/snap/foo_bar -> /var/snap/foo
+	fmt.Fprintf(&buf, "  mount options=(rw rbind) /var/snap/%s/ -> /var/snap/%s/,\n", si.InstanceName(), si.SnapName())
+	// /home/joe/snap/foo_bar -> /home/joe/snap/foo
+	// workaround https://bugs.launchpad.net/ubuntu/+source/apparmor/+bug/1612393
```

Comment:
I'm sure @jdstrand would love to see this inside the generated profile, not just in the source code here, as this makes the auditing process easier.

Reply:
For reference:
```
$ cat /var/lib/snapd/mount/snap.test-snapd-tools_foo.fstab 
/snap/test-snapd-tools_foo /snap/test-snapd-tools none rbind,x-snapd.origin=parallel-instance 0 0
/var/snap/test-snapd-tools_foo /var/snap/test-snapd-tools none rbind,x-snapd.origin=parallel-instance 0 0
$ cat /var/lib/snapd/mount/snap.test-snapd-tools_foo.user-fstab 
$USER_HOME_SNAP_DIR/test-snapd-tools_foo $USER_HOME_SNAP_DIR/test-snapd-tools none rbind,x-snapd.origin=parallel-instance 0 0
```

Reply:
Yeah, I meant the workaround comment.